src/main/java/gov/nasa/pds/harvest/dao/EsRequestBuilder.java

```diff
@@ -57,7 +57,8 @@ private JsonWriter createJsonWriter(Writer writer)
      */
     public String createSearchIdsRequest(Collection<String> ids, int pageSize) throws Exception
     {
-        if(ids == null || ids.isEmpty()) throw new Exception("Missing ids");
+        if(ids == null || ids.isEmpty()) throw new Exception("Error reading bundle/collection references. " +
```

Comment:
I see that the method already declares that it throws the generic `Exception`, but is there a more specific (more application-specific) exception class it can use?

Reply:
@nutjob4life and @jordanpadams, Thank you!

I also thought of this, but noticed that there are many places in this codebase that use Generic Exception and planned to do a restructuring later (since we plane to combine harvest/registry-mgr/supplementer into one registry-loader repo.

However, I added a new Exception called InvalidPDS4ProductException and used it here.

Reply:
👍